### PR TITLE
🐛 Source Sentry: fix `None` state_value + other bad `state_values`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/oncall/issues/1575

## How
- added dedicated `get_state_value` method to parse the state correctly

## 🚨 User Impact 🚨
No impact is expected.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-sentry/source_sentry/streams.py [hunk 1]
```diff
@@ -69,14 +71,21 @@ def __init__(self, *args, **kwargs):
         super(SentryIncremental, self).__init__(*args, **kwargs)
         self._cursor_value = None
 
+    def get_state_value(self, stream_state: Mapping[str, Any] = None) -> str:
+        bad_state_values = [None, "None", ""]
+        state = self.start_date
+        if stream_state:
+            state_value = stream_state.get(self.cursor_field)
+            state = state if state_value in bad_state_values else state_value
+        return pendulum.parse(state)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-sentry/source_sentry/streams.py:80]: Better not to hardcode bad values. 


```suggestion
    def validate_state_value(self, state_value) -> str:   
        none_as_string_or_empty = state_value == "None" if state_value else True
        return self.start_date if none_as_string_or_empty else state_value


    def get_state_value(self, stream_state: Mapping[str, Any] = None) -> str:
        state_value = self.validate_state_value(stream_state.get(self.cursor_field, self.start_date) if stream_state else self.start_date)
        return pendulum.parse(state_value)
```

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-sentry/source_sentry/streams.py [hunk 2]
```diff
@@ -20,6 +20,8 @@ class SentryStream(HttpStream, ABC):
     def __init__(self, hostname: str, **kwargs):
         super().__init__(**kwargs)
         self._url_base = self.URL_TEMPLATE.format(hostname=hostname, api_version=self.API_VERSION)
+        # hardcode the start_date default value, since it's not present in spec.
+        self.start_date = "1900-01-01T00:00:00.0Z"
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte-integrations/connectors/source-sentry/source_sentry/streams.py:24]: The `input_configuration` doesn't have the `start_date`  field. I believe it was hardcoded before this PR, I've just moved it to another place.

File: airbyte-integrations/connectors/source-sentry/source_sentry/streams.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-sentry/source_sentry/streams.py:80]: Love it.

File: airbyte-integrations/connectors/source-sentry/source_sentry/streams.py [hunk 3]
```diff
@@ -69,14 +71,21 @@ def __init__(self, *args, **kwargs):
         super(SentryIncremental, self).__init__(*args, **kwargs)
         self._cursor_value = None
 
+    def validate_state_value(self, state_value: str = None) -> str:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-sentry/source_sentry/streams.py:74]: Let's add unit test for this two methods below 😅

**Review by lazebnyi — APPROVED** [review]
LGTM! Only need add unit tests 😊

File: airbyte-integrations/connectors/source-sentry/source_sentry/streams.py (hunk 3 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-sentry/source_sentry/streams.py:74]: Done

**Pull request merged**